File: replace_ai.js
const fs = require('fs');
const path = require('path');

const directory = path.join(__dirname, 'src');

function replaceInFile(filepath) {
    try {
        const content = fs.readFileSync(filepath, 'utf8');
        let newContent = content.replace(/AI Building/g, 'New Academic Block');
        newContent = newContent.replace(/AI Department/g, 'New Academic Block');
        
        if (newContent !== content) {
            fs.writeFileSync(filepath, newContent, 'utf8');
            console.log(`Updated ${filepath}`);
        }
    } catch (e) {
        console.error(`Error processing ${filepath}:`, e);
    }
}

function walkWalk(dir) {
    const list = fs.readdirSync(dir);
    for (const file of list) {
        const filepath = path.join(dir, file);
        const stat = fs.statSync(filepath);
        if (stat && stat.isDirectory()) {
            walkWalk(filepath);
        } else if (file.endsWith('.ts') || file.endsWith('.tsx')) {
            replaceInFile(filepath);
        }
    }
}

walkWalk(directory);
console.log('Done');

File: replace_ai.py
import os

directory = r"c:\Users\hp\Desktop\mitsvirtual2\src"

def replace_in_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
    except Exception as e:
        print(f"Error reading {filepath}: {e}")
        return

    new_content = content.replace("AI Building", "New Academic Block")
    new_content = new_content.replace("AI Department", "New Academic Block")
    
    if new_content != content:
        try:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f"Updated {filepath}")
        except Exception as e:
            print(f"Error writing {filepath}: {e}")

for root, _, files in os.walk(directory):
    for file in files:
        if file.endswith(('.ts', '.tsx')):
            replace_in_file(os.path.join(root, file))

print("Done")
